### ISSUE TITLE
[BUG] Partner Sharing - Unable to delete photo

### ISSUE THREAD

**Issue by yurividal** [immich-app/immich#5213]
### The bug

When partner sharing on the time-line is enabled, the user who doesn't own the picture cannot delete a shared picture. The delete button does nothing.
If this is by design, then the button needs to be hidden, or some sort of UI element to tell the user he can't delete partner's photos.
Even better would be if there was an option in the settings to allow a user to do such a thing.

### The OS that Immich Server is running on

Ubuntu 22.04

### Version of Immich Server

V1.88.1

### Version of Immich Mobile App

V1.88.1

### Platform with the issue

- [X] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Same as example one.
```

### Your .env content

```Shell
Same as example one
```

### Reproduction steps

```bash
1. Enable partner sharing
2. Enable shared photos on timeline
3. Try to delete a photo that belongs to your partner. Mobile and Web
...
```

**Comment by BarrettLowe** [comment]
Looks like the following may be the culprit:
https://github.com/immich-app/immich/blob/8a2b36ad55169a6befe4a23bfac8e4f8b697af43/web/src/lib/components/photos-page/actions/delete-assets.svelte#L40

I don't know typescript so I'm having a hard time tracking this backward but at an initial glance, it may work to change this to use getAssets instead...?

**Comment by yurividal** [comment]
> This is as designed. Let me think of a way to make it less confused

Thought it would be as design. Indeed a UI hint is needed here.

Even better would be if there was a option to allow deletion of partners photos.

(Example: my wife is constantly taking pictures of receipts or things she only needs for a few minutes, and never remembers to delete them from immich. I'd love to be able to clean these every now and then, instead of filling our precious storage space with useless pictures)

**Comment by Fl0ete** [comment]
> > This is as designed. Let me think of a way to make it less confused
> 
> Thought it would be as design. Indeed a UI hint is needed here.
> 
> Even better would be if there was a option to allow deletion of partners photos.
> 
> (Example: my wife is constantly taking pictures of receipts or things she only needs for a few minutes, and never remembers to delete them from immich. I'd love to be able to clean these every now and then, instead of filling our precious storage space with useless pictures)

I'd also love to see this. In my case, I am always the one "thinning out" the photos, deleteing blurred ones, keeping only the best shot of many of the same motive. And I would love to just show my partners photos in my timeline (especially with kids), but having them raw is not really satisfying.

Kind of a big ask, but for me the optimal solution would be something like this: I mark the pictures for deletion and my partner gets a notification/hint to look over them and confirm. This way, she doesn't have the work and can still be sure I delete nothing important to her.

**Comment by Jab2870** [comment]
I'm not sure if this is really the correct place to ask, but I would really like to see the partner sharing to allow a lot more control.

Deleting photos is certainly part of that, but other stuff like being able to add a partner's photos to an album etc.

**Comment by andrewdcaudle** [comment]
I concur with a setting that would allow partners to delete the other's images.

**Comment by Tonylolo** [comment]
Yes , very good idea, partner would become a real user. 2 users of 1 photo server, a lot of case for family i think.
Thk for taking in consideration

**Comment by krin-san** [comment]
> I concur with a setting that would allow partners to delete the other's images.

I actually tried to find if this option exists on `/user-settings?isOpen=partner-sharing` page. Imagined, it could look like that:

![Screen Shot 2024-04-19 at 00 01 47](https://github.com/immich-app/immich/assets/2318002/c83d7a2e-9864-435f-a043-ec10d65657f8)
